velox/vector/arrow/Bridge.cpp

```diff
@@ -334,10 +345,72 @@ struct Selection {
   vector_size_t total_;
 };
 
+// Gather values from timestamp buffer. Nulls are skipped.
+void gatherFromTimestampBuffer(
+    const BaseVector& vec,
+    const Selection& rows,
+    TimestampUnit unit,
+    Buffer& out) {
+  auto src = (*vec.values()).as<Timestamp>();
+  auto dst = out.asMutable<int64_t>();
+  if (!vec.mayHaveNulls() || vec.getNullCount() == 0) {
+    switch (unit) {
+      case TimestampUnit::kSecond:
+        rows.apply([&](vector_size_t i) { dst[i] = src[i].getSeconds(); });
```

Comment:
@rui-mo we can not just set data to dst[i], i is not start from 0 if the offset in `rows` is not zero.

Reply:
@ViggoC Thank you so much for catching this. Filed https://github.com/facebookincubator/velox/pull/9170 to fix this issue. Would you help take a review?